File: backend/issue-service/src/main/java/com/example/issueservice/dto/rabbit/AttachmentDeletedEvent.java
package com.example.issueservice.dto.rabbit;

import com.example.issueservice.dto.models.Attachment;

import java.time.Instant;

public record AttachmentDeletedEvent(
        long projectId,
        long issueId,
        long attachmentId,
        long deleterId,
        Instant deletedAtUtc
) {
    public static AttachmentDeletedEvent from(Attachment attachment, long projectId) {

        return new AttachmentDeletedEvent(
                projectId,
                attachment.getIssue().getId(),
                attachment.getId(),
                attachment.getCreatedBy(),
                Instant.now()
        );
    }
}

File: backend/dashboard-service/Backend.Dashboards.Api/Dto/AttachmentDeletedEvent.cs
using System.Text.Json.Serialization;

namespace Backend.Shared.DTOs;

public class AttachmentDeletedEvent
{
    [JsonPropertyName("projectId")]
    public long ProjectId { get; set; }

    [JsonPropertyName("issueId")]
    public long IssueId { get; set; }

    [JsonPropertyName("attachmentId")]
    public long AttachmentId { get; set; }

    [JsonPropertyName("deleterId")]
    public long DeleterId { get; set; }

    [JsonPropertyName("DeletedAtUtc")]
    public DateTime DeletedAtUtc { get; set; } = DateTime.UtcNow;
}